backend/api/views.py

```diff
@@ -190,6 +191,75 @@ def experiment_unenrolls(request, exp_slug):
     return Response(data)
 
 
+@api_view(['GET'])
+def experiment_enrolls_full(request, exp_slug):
+    enrollment = Enrollment.objects.filter(experiment=exp_slug).exists()
+    if not enrollment:
+        raise NotFound('No experiment with given slug found.')
+
+    branches = list(
+        Enrollment.objects.filter(experiment=exp_slug)
+                          .exclude(branch__isnull=True)
+                          .distinct('branch')
+                          .values_list('branch', flat=True)
+    )
```

Comment:
Try doing what was done in the population view with the `not in` check to add the branch and array instead of doing this extra query.

Reply:
Good call though not sure if they were included in the dict...I'll see what I can do.